Kitodo/src/main/java/org/kitodo/production/filters/FilterMenu.java

```diff
@@ -296,6 +297,12 @@ public List<ParsedFilter> getParsedFilters() {
      * @return value of filterInEditMode
      */
     public String getFilterInEditMode() {
+        Map<String, String> parameters = FacesContext.getCurrentInstance().getExternalContext()
+                .getRequestParameterMap();
+        if (parameters.containsKey("input") && StringUtils.isBlank(filterInEditMode)) {
+            filterInEditMode = parameters.get("input");
```

Comment:
You are taking over any argument from the outside without validating that the given value is correct or at least is in range of the expectations?

Reply:
Yes, the user can enter any string into the search slot. The input is not “checked”. But of course, if the user has entered nonsense, the search will not find any results.

Reply:
Thank you for this clarification but I expected something different as I read the method, method content  and the used written class variable `filterInEditMode`. Handling of this class variable `filterInEditMode` is strange in this class but this is a different story.